algoliasearch/algoliasearch.go

```diff
@@ -9,6 +9,9 @@ import (
 // allows manipulations over the indexes of the application as well as network
 // related parameters.
 type Client interface {
+	// GetAppID returns the Algolia application ID to which the client is linked to.
+	GetAppID() string
```

Comment:
This is breaking, we already had this discussing before. Just wanted it again, in order to consider move this Pull Request to the go v3.

Reply:
As discussed, using an interface was a bad idea. Without changing the interface, I can't really touch any part of the public API (add/delete/modify methods). So for the current major version of the client, decision was made to let us break the compatibility for added methods only.

However, I definitely hear your point on compatibility and we'll make sure to do differently in the next major release.